### ISSUE TITLE
[a11y]: ContainedList - [with expandable & persistent search] - misleading search label

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13944]
### Package
@carbon/react

### Browser
Safari & FF & iPhone

### Package version
carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

2.4.6 Headings and Labels 

### Assistive technology
JAWS & FF

### Description
Search button and search edit field labels are misleading 

### WCAG 2.1 Violation
2.3.4 Headings and Labels

### Reproduction/example
Record `Number 32`  in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-containedlist--with-expandable-search

### Steps to reproduce
1. The label for the search button is misleading for the user. The user may think that the button triggers the action of search up on activation.

2. The label "search" does not convey the purpose of the search edit field. which is to filter the options present after it.

### User impact: 
If the label is incorrect or inadequately descriptive, it affects the screen reader user’s ability to understand the purpose of the control.   

### Recommended fix: 
1. The label can be given as "open search"
2. The label can be given as "filter search"
3. For the benefit of the SR user following behavior can be implemented if possible.

A. Provide instruction saying that "the options present after the search field will be modified based on the input". And associate it using `aria-describedby`.
**Extracting the below part of the issue into it's separate issue here:** https://github.com/carbon-design-system/carbon/issues/14103
B. Make the screen reader announce the dynamic change of results instantaneously.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
